lib/workers/branch/get-updated.ts

```diff
@@ -260,16 +260,26 @@ export async function getUpdatedPackageFiles(
         config,
       });
       if (is.nonEmptyArray(results)) {
-        updatedPackageFiles.push(packageFile);
+        let updated = false;
         for (const res of results) {
           const { file, artifactError } = res;
           // istanbul ignore else
-          if (file) {
+          if (
+            file &&
+            (await getFile(
+              file.name,
+              reuseExistingBranch ? config.branchName : config.baseBranch
+            )) !== file.contents
```

Comment:
Wait, the contents can be a buffer (Gradle wrapper) or directory (git submodule).

So that needs to be handled too.